src/unstract/sdk/index.py

```diff
@@ -388,14 +395,25 @@ def generate_file_id(
         index_key = {
             "tool_id": tool_id,
             "file_hash": file_hash,
-            "vector_db_config": ToolAdapter.get_adapter_config(self.tool, vector_db),
-            "embedding_config": ToolAdapter.get_adapter_config(self.tool, embedding),
-            "x2text_config": ToolAdapter.get_adapter_config(self.tool, x2text),
             # Typed and hashed as strings since the final hash is persisted
             # and this is required to be backward compatible
             "chunk_size": str(chunk_size),
             "chunk_overlap": str(chunk_overlap),
         }
+
+        if not self.is_public_call:
```

Comment:
@tahierhussain what happens to the doc_id when we decide to use different adapters for this simple prompt studio feature? I think the config / JSON itself should be used to obtain the unique ID

Reply:
@chandrasekharan-zipstack As per our earlier discussion, this scenario will now be handled with the latest changes.